phenotrex/ml/prediction.py

```diff
@@ -58,9 +59,14 @@ def predict(
     model = load_classifier(filename=classifier, verb=verb)
     if out_explain_per_sample is not None or out_explain_summary is not None:
         try:
-            fs, sv, bv = model.get_shap(gr, n_samples=shap_n_samples)
+            fs, sv, bv = model.get_shap(
+                gr, n_samples=shap_n_samples, n_features=n_max_explained_features
+            )
         except TypeError:
             raise RuntimeError('This TrexClassifier is not capable of generating SHAP explanations.')
+        except MemoryError:
+            os._exit(137)  # exit immediately with catchable exit code
+            raise RuntimeError
```

Comment:
I suppose this is relevant for a pipeline around phentrex? How about the following, which wouldn't swallow the informative error message about too high memory requirement:
```suggestion
        except MemoryError as err:
            os._exit(137)  # exit immediately with catchable exit code
            raise RuntimeError from err
```

Reply:
fair enough. I actually just added the final raise to stop pycharm from bugging me about broken control flow / uninitialized variables (it evidently doesn't know that after os._exit(), everything dies :D), but this is more formally correct for sure. probably I'll just reraise the MemoryError.

Reply:
Makes total sense! I updated the code accordingly.

Reply:
Well on the other hand, since this error is then never actually raised, perhaps we could send out a warning/error before `os._exit`?